### PR TITLE
Depth average ascii text output

### PR DESCRIPTION
**Pull request by ian-r-rose**
I have found this to be useful for the DepthAverage postprocessor.  Outputting the results as a finite element space can make the resulting files difficult to read and process (especially with standard unix text parsing tools). Furthermore, these output files are unlikely to be so large as to be expensive, from both and I/O standpoint and a disk usage standpoint.

This adds the option to print lateral averages as a plain ascii text table. It does not change existing functionality.

### CODE REVIEW

File: source/postprocess/depth_average.cc [hunk 1]
```diff
@@ -170,55 +171,85 @@ namespace aspect
...
+                }
+              else f.open(filename.c_str(), std::ofstream::app);
 
+              double depth = max_depth/double(data_point.values[0].size())/2.0;
+              for (unsigned int d = 0; d < data_point.values[0].size(); ++d)
+                {
+                  f <<std::setw(12)<< depth <<" "<<std::setw(12)<<
+                    (this->convert_output_to_years() ? data_point.time/year_in_seconds : data_point.time)<<" ";
+                  for ( unsigned int i = 0; i < variables.size(); ++i )
+                    f << std::setw(12)<< data_point.values[i][d]<<" ";
```
Comment (bangerth) [inline source/postprocess/depth_average.cc]: Here and elsewhere, can you consistently put a space around `<<`?

**Comment by bangerth** [comment]
Looks pretty good overall.

**Comment by tjhei** [comment]
For gnuplot compatibility it would be great to:
./ start the header with "#"
./ put empty lines between different times

Do you feel strongly about the order of the columns? I would put the time first.

Oh, and didn't we use to have ascii depth-averaging and you ripped it out, @bangerth ? :-)

File: source/postprocess/depth_average.cc [hunk 2]
```diff
@@ -170,55 +171,85 @@ namespace aspect
...
+                  f << "depth       " << "time        ";
+                  for ( unsigned int i = 0; i < variables.size(); ++i)
+                    {
+                      f << variables[i] << " ";
+                    }
+                  f << std::endl;
+                }
+              else f.open(filename.c_str(), std::ofstream::app);
 
+              double depth = max_depth/double(data_point.values[0].size())/2.0;
```
Comment (tjhei) [inline source/postprocess/depth_average.cc]: please use c++ style casts: ```static_cast<double>(...)```

File: source/postprocess/depth_average.cc [hunk 3]
```diff
@@ -170,55 +171,85 @@ namespace aspect
...
+              for (unsigned int d = 0; d < data_point.values[0].size(); ++d)
+                {
+                  f << std::setw(12) << depth << " " << std::setw(12) <<
+                    (this->convert_output_to_years() ? data_point.time/year_in_seconds : data_point.time) << " ";
+                  for ( unsigned int i = 0; i < variables.size(); ++i )
+                    f << std::setw(12) << data_point.values[i][d] << " ";
+                  f << std::endl;
+                  depth+= max_depth/double(data_point.values[0].size() );
```
Comment (tjhei) [inline source/postprocess/depth_average.cc]: same here

**Comment by tjhei** [comment]
I think it makes sense to write the file from scratch each time (we have all data available anyways). Otherwise your output will be weird if you resume from a snapshot.

**Comment by ian-r-rose** [comment]
I would probably put time first as well, but the gnuplot output has time as the "y" coordinate.

Suspend/resume is a good point. I will take a crack at rewriting at every depth-average tomorrow.

**Comment by tjhei** [comment]
> I would probably put time first as well, but the gnuplot output has time as the "y" coordinate.

You mean if you currently pick "gnuplot" as the output format (and not ascii)? I would say that doesn't matter.

**Comment by ian-r-rose** [comment]
Fair enough, I am perfectly happy to put time first.

**Comment by bangerth** [comment]
@ian-r-rose -- the fact that the DataOutStack format puts time second should not constrain you. Do as you think is appropriate. (DataOutStack treats the data in a way so that it can be plotted as a space-time diagram.)

@tjhei -- I didn't mean to rip it out, but rather wanted to generalize it in a way so that it can be shown as a space-time diagram using any of the usual visualization tools. That seemed like the most appropriate direction at a time. The intent wasn't to purposefully not do what @ian-r-rose wants to do.

**Comment by tjhei** [comment]
@bangerth no worries!

@ian-r-rose, I realized we already have gnuplot output so do whatever works best for you and ignore my comments. Are you writing it to be read in with numpy.loadtxt?

**Comment by ian-r-rose** [comment]
I would like to be able to use numpy.loadtxt() or just awk/paste/etc.

I have swapped the time and depth columns, and now I regenerate the txt file every time the depth average is computed, so suspend/resume should work.

**Comment by ian-r-rose** [comment]
Anything else to be done here? This is fairly ready from my perspective.

**Comment by tjhei** [comment]
nope! :+1:

**Pull request merged**